File: cpp/ISP/step2/Vehicle.cpp
#pragma once
#include "Switches.cpp"

class Vehicle : public Switches {
public:
	bool IsEngineRunning() {
		return engine_running_;
	}

	void StartEngine() override {
		engine_running_ = true;
	}

	void ShutDownEngine() override {
		engine_running_ = false;
	}

private:
	bool engine_running_;
};

File: python/ISP/step2/vehicle.py
from switches import Switches

class Vehicle(Switches):
    def __init__(self) -> None:
        super().__init__()
        self.__engin_running = False

    def is_engin_running(self) -> bool:
        return self.__engin_running

    def start_engine(self):
        self.__engin_running = True

    def shutdown_engine(self):
        self.__engin_running = False

